operator/pkg/central/extensions/reconcile_tls_test.go

```diff
@@ -172,6 +260,55 @@ func TestCreateCentralTLS(t *testing.T) {
 	}
 }
 
+func gettingSecretFails(secretName string) testutils.InterceptorFns {
+	return testutils.InterceptorFns{
+		Get: func(ctx context.Context, client ctrlClient.WithWatch, key ctrlClient.ObjectKey, obj ctrlClient.Object, opts ...ctrlClient.GetOption) error {
+			_, ok := obj.(*v1.Secret)
+			if !ok || key.Name != secretName {
+				return client.Get(ctx, key, obj, opts...)
+			}
+			return k8sErrors.NewServiceUnavailable("failure")
+		},
+	}
+}
+
+func creatingSecretFails(secretName string) testutils.InterceptorFns {
+	return testutils.InterceptorFns{
+		Create: func(ctx context.Context, client ctrlClient.WithWatch, obj ctrlClient.Object, opts ...ctrlClient.CreateOption) error {
+			secret, ok := obj.(*v1.Secret)
+			if !ok || secret.Name != secretName {
+				return client.Create(ctx, obj, opts...)
+			}
+			return k8sErrors.NewServiceUnavailable("failure")
+		},
+	}
+}
+
+func deletingSecretFails(secretName string) testutils.InterceptorFns {
+	return testutils.InterceptorFns{
+		Delete: func(ctx context.Context, client ctrlClient.WithWatch, obj ctrlClient.Object, opts ...ctrlClient.DeleteOption) error {
+			secret, ok := obj.(*v1.Secret)
+			if !ok || secret.Name != secretName {
+				return client.Delete(ctx, obj, opts...)
+			}
+			return k8sErrors.NewServiceUnavailable("failure")
```

Comment:
I think it would read easier if the condition was flipped: "if this is what we're looking for, return error; otherwise do the delete". Similarly for the other cases.

Reply:
Agreed